packages/container/lib/container.js

```diff
@@ -350,7 +474,7 @@ function instantiate(container, fullName) {
 
     if (typeof factory.extend === 'function') {
```

Comment:
This refers to `extend` on the registered class, not `extend` on the factory object. The idea being `extend` is supported, injections have been made.

`instantiate` is only called by `lookup` though, and in that case we want the double extend to continue for now so this test would hold true.